blob/gcsblob/gcsblob_test.go

```diff
@@ -143,9 +143,17 @@ func (verifyContentLanguage) ErrorCheck(b *blob.Bucket, err error) error {
 }
 
 func (verifyContentLanguage) BeforeWrite(as func(interface{}) bool) error {
+	var objp **storage.ObjectHandle
+	if !as(&objp) {
+		return errors.New("Writer.As failed to get ObjectHandle")
+	}
+	// Replace the ObjectHandle with a new one that adds Conditions
+	// (this condition will always be true).
+	*objp = (*objp).If(storage.Conditions{DoesNotExist: true})
```

Comment:
I think we should make this a separate test case: it seems like making the precondition fail is the test case that would  ensure that this escape hatch worked. Otherwise, if it's always true, then it would be the same as if the escape hatch didn't modify the outgoing request.

Reply:
Done.